src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorPicker.cs

```diff
@@ -1184,6 +1094,167 @@ private Rgb ApplyConstraintsToRgbColor(Rgb rgb)
 			return ColorConversion.HsvToRgb(hsv);
 		}
 
+		private CompositeDisposable SubscribeToEvents()
```

Comment:
I have no experience with WASM; however, what is different about the lifecycle of a control to make this necessary? Or it is a larger Uno issue that applies to other platforms I am unaware of?

Reply:
So problem here is that `_eventSubscritions` are disposed on `Unloaded` event. Now without re-subscribing them on reloaded, would result in the control becomes completely irresponsive after unloaded and then reloaded, which is the case when the control is placed inside a popup, and had its `IsOpen` toggled (3rd item in #5992).

Reply:
Oh, this makes more sense in a Flyout/Popup. I can understand that things will be unloaded when the flyout collapses in some cases. That said, WinUI code itself works fine in a Flyout so clearly Uno is handling this differently. As Uno tries to behave like WinUI, I don't think Uno should actually be unloading the control here. Are we sure this is correct control lifecycle in this case? Is Uno being a bit overzealous reclaiming resources when it shouldn't be? (this might be related to recent optimizations as well)

It seems to be this is a work-around for an Uno Specific issue.

Reply:
Ok, nevermind, I understand now what you are saying. I added in the `Unloaded` event code that explicitly removes events -- which is clearly a bad idea. I wonder if this code should be removed completely:

https://github.com/unoplatform/uno/blob/9f2b080f8d009664153404032fe58889a698bd37/src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorPicker.cs#L724-L725

Will this cause an issue on full AOT platforms like iOS? I would expect the events will be properly cleaned up as the ColorPicker itself is freed. Maybe finalizers can be used instead as well?

```csharp
~ColorPicker()
{
  // Uno Doc: Added to dispose event handlers 
   _eventSubscriptions.Disposable = null; 
}
```

I have no experience with how Mono runtime and AOT will handle things here so these are just ideas. It would be nice not to change the code so drastically from WinUI though.

Reply:
It's generally important to unsubscribe cross-control event registrations on unloading, particularly on iOS. Otherwise it leaves a circular reference which creates a [memory leak](https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/performance#avoid-strong-circular-references). It's necessary to deviate from the WinUI code in this case since the underlying system comportment is different.

Reply:
Alright, makes sense. Was hoping there was something that could be done differently but it is what it is. Thanks for explaining